modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -88,11 +88,42 @@ export function getValueDetailsFromItem(item, fieldName) {
 		return null;
 	}
 
+	// TODO: extract function to get same behavior as block 130-143
+	const languageId =
+		Liferay.ThemeDisplay.getLanguageId() ||
+		Liferay.ThemeDisplay.getBCP47LanguageId() ||
+		Liferay.ThemeDisplay.getDefaultLanguageId();
+
+	const rawTextFieldName = `${fieldName}RawText`;
+	const i18nFieldName = `${fieldName}_i18n`;
+
 	let rootPropertyName = fieldName;
 	const valuePath = [];
 	let navigatedValue = item;
 
-	if (Array.isArray(fieldName)) {
+	// Cover richText content value
+	// {name: '<p>...</p>', nameRawText: '...'}
+	if (
+		Object.hasOwn(item, rawTextFieldName) &&
+		Object.values(rawTextFieldName).length
+	) {
+		valuePath.push(fieldName);
+		navigatedValue = navigatedValue[rawTextFieldName];
+	}
+	// Cover _i18n fields with language keys
+	// {name: '', name_i18n: {en_US: ''}}
+	else if (Object.hasOwn(item, i18nFieldName) &&
```

Comment:
This part covers the bug where the non-localized field comes with a wrong value (see https://liferay.atlassian.net/browse/LPS-194819) . Does not hurt but in the end, this not should be here.